ui/rtk-query/user.js

```diff
@@ -18,6 +19,24 @@ const userApi = api.injectEndpoints({
       query: () => 'user/prefs',
       method: 'GET',
     }),
+    updateUserPref: builder.mutation({
+      query: (queryArg) => ({
+        url: 'user/prefs',
+        method: 'POST',
+        body: queryArg,
+        credentials: 'include',
+      }),
+    }),
+    updateUserPrefWithContext: builder.mutation({
+      query: (queryArg) => ({
+        url: ctxUrl('/user/prefs', queryArg.selectedK8sContexts),
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json;charset=UTF-8',
+        },
+        body: queryArg,
```

Comment:
@senthil-athiban for clarity and consistency pass the body based on the querArg
```suggestion
        body: queryArg.body,
```
and pass the payload with respect to body. See other places